### PR TITLE
bpo-33083 - math.factorial accepts non-integral Decimal instances

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue33083

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2018-03-19-00-59-20.bpo-33083.Htztjl.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+math.factorial no longer accept arguments that are not int-like. A
```
Comment (DylanMeeus) [inline Misc/NEWS.d/next/Core and Builtins/2018-03-19-00-59-20.bpo-33083.Htztjl.rst]: Sorry for nit-picking a bit. But 

"math.factorial no longer accepts arguments.." (accept -> accepts).

emmited -> emitted

File: Modules/mathmodule.c [hunk 2]
```diff
@@ -1671,9 +1671,21 @@ math_factorial(PyObject *module, PyObject *arg)
...
+        if (PyErr_WarnEx(PyExc_DeprecationWarning,
+                "Passing float instances to math.factorial will be "
+                "removed in future versions.", 1) < 0) {
+            return NULL;
+        }
+
+    }
+    else{
+        pyint_form = PyNumber_Index(arg);
+        if( pyint_form == NULL){
```
Comment (taleinat) [inline Modules/mathmodule.c]: Fix the spacing on this and the next line.

**Review by taleinat — CHANGES_REQUESTED** [review]
It was decided on the issue to avoid deprecating floats as part of this issue. Please undo the changes regarding deprecation of floats; though perhaps keep them stashed somewhere for future use...

**Comment by pablogsal** [comment]
I have made the requested changes; please review again

File: Misc/NEWS.d/next/Core and Builtins/2018-03-19-00-59-20.bpo-33083.Htztjl.rst (hunk 1 above)
Comment (pablogsal) [reply Misc/NEWS.d/next/Core and Builtins/2018-03-19-00-59-20.bpo-33083.Htztjl.rst]: Thanks! :)

**Review by taleinat — CHANGES_REQUESTED** [review]
The NEWS entry no longer correctly reflects the changes made; please revise.

**Comment by pablogsal** [comment]
I have made the requested changes; please review again. Thank you for the whitespace fix!

**Review by taleinat — APPROVED** [review]
LGTM

**Review by serhiy-storchaka — APPROVED** [review]

**Review by mdickinson — APPROVED** [review]
LGTM

**Review by vstinner — COMMENTED** [review]
LGTM, you can merge it @pablogsal ;-)

**Review by vstinner — APPROVED** [review]
LGTM. @pablogsal you can merge it.

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**